src/posit/connect/content.py

```diff
@@ -137,12 +140,231 @@ class ContentItem(Resource):
+        url = urls.append(self.config.url, f"v1/content/{self.guid}")
+        response = self.session.patch(url, json=body)
+        super().update(**response.json())
+
+    def _re_whatever(self) -> Task | None:
+        """Submit a re-whatever request (i.e., restart, refresh, etc).
+
+        A re-whatever is a catch-all term for restarting, refreshing, or re-whatever-ing the content requires to bounce it to a new state.
+
+        refresh:
+            For content that require variants. Find the default variant and render it again.
+
+        restart:
+            For content that require server threads. Toggle an unique environment variable and open the content, which activates a new server thread.
+
+        Returns
+        -------
+        Task | None:
+            A task for the content render when available, otherwise None
+
+        Raises
+        ------
+        RuntimeError
+            Found an incorrect number of default variants.
+
+        Examples
+        --------
+        >>> _re_whatever()
+        """
+        # Update the item to its current state.
+        # The 'app_mode' is not set until a bundle is created and deployed.
+        # During the deployment process, the 'app_mode' is read from manifest.json and written to the database.
+        # Until this occurs the 'app_mode' will be 'unknown'.
+        self.update()
+
+        if self.app_mode in {
+            "rmd-static",
+            "jupyter-static",
+            "quarto-static",
+        }:
+            variants = self._variants.find()
+            variants = [variant for variant in variants if variant.is_default]
+            if len(variants) != 1:
+                raise RuntimeError(
+                    f"Found {len(variants)} default variants. Expected 1. Without a single default variant, the content cannot be refreshed. This is indicative of a corrupted state."
+                )
+            variant = variants[0]
+            return variant.render()
+
+        if self.app_mode in {
+            "api",
+            "jupyter-voila",
+            "python-api",
+            "python-bokeh",
+            "python-dash",
+            "python-fastapi",
+            "python-shiny",
+            "python-streamlit",
+            "quarto-shiny",
+            "rmd-shiny",
+            "shiny",
+            "tensorflow-saved-model",
+        }:
+            random_hash = secrets.token_hex(32)
+            key = f"_CONNECT_RESTART_TMP_{random_hash}"
+            self.environment_variables.create(key, random_hash)
+            self.environment_variables.delete(key)
+            # GET via the base Connect URL to force create a new worker thread.
+            url = urls.append(dirname(self.config.url), f"content/{self.guid}")
+            self.session.get(url)
+            return None
+
+        warnings.warn(
+            f"Content '{self.guid}' with application mode '{self.app_mode}' does not require restarts"
+        )
+        return None
```

Comment:
This is also new.

Reply:
In the R implementation, the already-implemented `render` app modes return an existing task object, so it's easy to return it. Do you think this is a reasonable point to diverge?

In R, I had moved towards just having the `render` function return a task object, but I'm not sure if consistency is better, particularly because interactive and rendered content are not different types.